interface/ceed-elemrestriction.c

```diff
@@ -102,6 +102,32 @@ int CeedElemRestrictionGetStrides(CeedElemRestriction rstr,
   return 0;
 }
 
+/**
+  @brief Get read-only access to a CeedElemRestriction offsets array by memtype
+
+  @param rstr         CeedElemRestriction to retrieve offsets
+  @param mtype        Memory type on which to access the array.  If the backend
+                        uses a different memory type, this will perform a copy
+                        (possibly cached).
+  @param[out] offsets Array on memory type mtype
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedElemRestrictionGetOffsets(CeedElemRestriction rstr, CeedMemType mtype,
+                                  const CeedInt **offsets) {
+  int ierr;
+
+  if (!rstr->GetOffsets)
+    // LCOV_EXCL_START
+    return CeedError(rstr->ceed, 1, "Backend does not support GetOffsets");
+  // LCOV_EXCL_STOP
+
+  ierr = rstr->GetOffsets(rstr, mtype, offsets); CeedChk(ierr);
+  return 0;
+}
+
```

Comment:
These being substantial data, should we have to return the pointers when we're done?

Reply:
Since this is access should be read only, is there an intended outcome if a user/backend does not return the pointer before destruction? We currently do not error for a `CeedVector` if read only access is never returned by the time the vector is destroyed. Is this a bug?

Reply:
I added a readers counter. I realized that it's possible to create a second restriction with a original restriction's offsets with `CEED_USE_POINTER`, destroy the first restriction, and break the second restriction.